ci/ReleaseTool/Common.cs

```diff
@@ -1,14 +1,80 @@
-using NLog;
+using Newtonsoft.Json.Linq;
 using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Text;
 using System.Text.RegularExpressions;
 
 namespace ReleaseTool
 {
     internal static class Common
     {
-        public const string RepoUrlTemplate = "git@github.com:{0}/{1}.git";
+        private const string RepoUrlTemplate = "git@github.com:{0}/{1}.git";
+
+        // Names of the version files that live in the UnrealEngine repository.
+        private const string UnrealGDKVersionFile = "UnrealGDKVersion.txt";
+        private const string UnrealGDKExampleProjectVersionFile = "UnrealGDKExampleProjectVersion.txt";
+
+        // Plugin file configuration.
+        private const string PluginFileName = "SpatialGDK.uplugin";
+        private const string VersionKey = "Version";
+        private const string VersionNameKey = "VersionName";
+
+        // Changelog file configuration
+        private const string ChangeLogFilename = "CHANGELOG.md";
+        private const string ChangeLogReleaseHeadingTemplate = "## [`{0}`] - {1:yyyy-MM-dd}";
+
+        // Unreal version configuration
+        private const string UnrealEngineVersionFile = "ci/unreal-engine.version";
+
+        public static bool UpdateVersionFilesWithEngine(GitClient gitClient, string gitRepoName, string versionRaw, string engineVersions, NLog.Logger logger, string versionSuffix = "")
+        {
+            return UpdateVersionFiles_Internal(gitClient, gitRepoName, versionRaw, logger, versionSuffix, engineVersions);
+        }
+
+        public static bool UpdateVersionFilesButNotEngine(GitClient gitClient, string gitRepoName, string versionRaw, NLog.Logger logger, string versionSuffix = "")
+        {
+            return UpdateVersionFiles_Internal(gitClient, gitRepoName, versionRaw, logger, versionSuffix);
+        }
+
+        private static bool UpdateVersionFiles_Internal(GitClient gitClient, string gitRepoName,  string versionRaw, NLog.Logger logger, string versionSuffix = "", string engineVersions = "")
+        {
+            string versionDecorated = versionRaw + versionSuffix;
+            switch (gitRepoName)
+            {
+                case "UnrealEngine":
+                {
+                    bool madeChanges = false;
+                    madeChanges |= UpdateVersionFile(gitClient, versionDecorated, UnrealGDKVersionFile, logger);
+                    madeChanges |= UpdateVersionFile(gitClient, versionDecorated, UnrealGDKExampleProjectVersionFile, logger);
+                    return madeChanges;
+                }
+                case "UnrealGDK":
+                {
+                    bool madeChanges = false;
+                    madeChanges |= UpdateChangeLog(gitClient, versionRaw);
```

Comment:
Don't we want to include the new UnrealGDKTestGymsVersion.txt file here already?
Saves us making another PR.

Reply:
Sure but that would be up in UnrealEngine case block right?

Reply:
Hm, no, I think the "UnrealGDK" block would be easier, since the GDK needs to know what versions of TestGyms to pull for its CI. We could make an exception and not have the version file in the top folder of the repo and move it into CI since it's only relevant there.

Reply:
I am not sure I love this spreading around of dependency tracking files

Reply:
Hm, it kind of makes sense in my head, as each repo is listing *its* own dependencies on other repos. Making it centralized would require CI to check the centralized repo. If it's the engine, that has the problem of having to check out the entire engine which is too costly. We can do what the EP CI does, which is check the GitHub webpage, but that is code that I am not proud to have written.

In the distant distant future, we may have a "SharedCI" repo where we could keep track of this, but currently we do not have it.

Reply:
agreed I'll move it for now. It's a bit of a mess : /